src/BinDeps.jl

```diff
@@ -364,26 +364,31 @@ module BinDeps
         ret
     end
 
-    @unix_only function lower(a::MakeTargets,collection)
-        cmd = `make -j8`
-        if(!isempty(a.dir))
-            cmd = `$cmd -C $(a.dir)`
-        end
-        if(!isempty(a.targets))
-            cmd = `$cmd $(a.targets)`
+    if is_unix()
+        function lower(a::MakeTargets,collection)
+            cmd = `make -j8`
+            if(!isempty(a.dir))
+                cmd = `$cmd -C $(a.dir)`
+            end
+            if(!isempty(a.targets))
+                cmd = `$cmd $(a.targets)`
+            end
+            @dependent_steps ( setenv(cmd, adjust_env(a.env)), )
         end
-        @dependent_steps ( setenv(cmd, adjust_env(a.env)), )
     end
-    @windows_only lower(a::MakeTargets,collection) = @dependent_steps ( setenv(`make $(!isempty(a.dir)?"-C "*a.dir:"") $(a.targets)`, adjust_env(a.env)), )
+    is_windows() && (lower(a::MakeTargets,collection) = @dependent_steps ( setenv(`make $(!isempty(a.dir)?"-C "*a.dir:"") $(a.targets)`, adjust_env(a.env)), ))
     lower(s::SynchronousStepCollection,collection) = (collection|=s)
 
     lower(s) = (c=SynchronousStepCollection();lower(s,c);c)
 
     #run(s::MakeTargets) = run(@make_steps (s,))
 
     function lower(s::AutotoolsDependency,collection)
-    	@windows_only prefix = replace(replace(s.prefix,"\\","/"),"C:/","/c/")
-    	@unix_only prefix = s.prefix
+        if is_windows()
+            prefix = replace(replace(s.prefix,"\\","/"),"C:/","/c/")
+        elseif is_unix()
+            prefix = s.prefix
+        end
     	cmdstring = "pwd && ./configure --prefix=$(prefix) "*join(s.configure_options," ")
```

Comment:
this should be tested on a package that uses AutotoolsDependency, but you likely need to restructure the above so `prefix = s.prefix` is set outside the if, then only on windows it gets overwritten